src/languages/bigquery.formatter.ts

```diff
@@ -835,8 +835,8 @@ export default class BigQueryFormatter extends Formatter {
       reservedBinaryCommands,
       reservedDependentClauses,
       reservedKeywords: dedupe([
-        ...Object.values(reservedFunctions).reduce((acc, arr) => [...acc, ...arr], []),
-        ...Object.values(reservedKeywords).reduce((acc, arr) => [...acc, ...arr], []),
+        ...Object.values(reservedFunctions).reduce((acc, arr) => acc.concat(arr), []),
+        ...Object.values(reservedKeywords).reduce((acc, arr) => acc.concat(arr), []),
```

Comment:
Why this whole "replace spread with concat"?

Reply:
- faster for large lists

Reply:
Whether there's any speed difference really comes down to how does a browser or NodeJS implement the spread operator. Ideally these run at the exact same speed, like Babel will compile the spread to concat when targeting platforms without spread support.

As always with performance, we should actually measure to see whether there is any practical difference in our case.

IMHO a larger problem is that we're repeating this `.reduce((acc, arr) => acc.concat(arr), [])` snippet all over. Essentially it's a `flatten()` function. Extracting a separate function for this would also provide better optimization opportunities if needed - like one could instead use a for loop instead of reduce.